server/player.lua

```diff
@@ -396,12 +396,26 @@ function QBCore.Player.CreatePlayer(PlayerData)
                         self.Functions.UpdatePlayerData()
                         TriggerEvent('qb-log:server:CreateLog', 'playerinventory', 'RemoveItem', 'red', '**' .. GetPlayerName(self.PlayerData.source) .. ' (citizenid: ' .. self.PlayerData.citizenid .. ' | id: ' .. self.PlayerData.source .. ')** lost item: [slot:' .. slot .. '], itemname: ' .. item .. ', removed amount: ' .. amount .. ', item removed')
                         return true
+                    elseif self.PlayerData.items[slot].amount < amountToRemove then
+                        amountToRemove = amountToRemove - self.PlayerData.items[slot].amount
+                        self.PlayerData.items[slot] = nil
+                        self.Functions.UpdatePlayerData()
+                        TriggerEvent('qb-log:server:CreateLog', 'playerinventory', 'RemoveItem', 'red', '**' .. GetPlayerName(self.PlayerData.source) .. ' (citizenid: ' .. self.PlayerData.citizenid .. ' | id: ' .. self.PlayerData.source .. ')** lost item: [slot:' .. slot .. '], itemname: ' .. item .. ', removed amount: ' .. amount .. ', item removed')
```

Comment:
Remove this, please

Reply:
![image](https://user-images.githubusercontent.com/54166713/155473645-ad8feb43-a9d8-4ae1-9816-fda9fc92e614.png)
![image](https://user-images.githubusercontent.com/54166713/155473664-e30dad13-3f6c-4884-b137-f0cffc4de1dc.png)

Can you try this?

Reply:
Why move all and return false? you're better off error-checking if you have the incorrect amount.

I see you did remove it though.